template/koa-rest-api/src/app.test.ts

```diff
@@ -10,11 +10,25 @@ describe('app', () => {
 
   it('has a happy health check', () => agent.get('/health').expect(200, ''));
 
-  it('has a reachable smoke test', () =>
-    agent.get('/smoke').expect(({ status }) => status !== 404));
+  it('has a reachable smoke test', async () => {
+    const response = await agent.get('/smoke');
+    expect(response.status).not.toBe(404);
+  });
 
+  /**
+   * Test assertion with logic
+   */
   it('has a reachable nested route', () =>
-    agent.get('/jobs').expect(({ status }) => status !== 404));
+    agent.get('/jobs').expect(({ status }) => {
```

Comment:
Can we do the same here? I'd prefer to avoid this `expect(fn)` form entirely given the mistake we previously made 🥲